File: packages/snow_draw_engine/lib/draw/edit/rotate/angle_calculator.dart
import '../../core/geometry/rotate_geometry.dart';
import '../../types/draw_point.dart';

double rawAngle({
  required DrawPoint currentPosition,
  required DrawPoint center,
}) => RotateGeometry.angleFromCenter(currentPosition, center);

/// Normalizes an angle delta to [-pi, pi].
double normalizeDelta(double delta) => RotateGeometry.normalizeDelta(delta);

/// Snaps a total angle (base + delta) to the nearest interval and returns
/// the snapped delta.
double applyDiscreteSnap({
  required double delta,
  required double baseAngle,
  required double snapInterval,
}) => RotateGeometry.applyDiscreteSnap(
  delta: delta,
  baseAngle: baseAngle,
  snapInterval: snapInterval,
);

File: packages/snow_draw_rust_engine/src/draw/edit/rotate/angle_calculator.rs
#![allow(dead_code)]

use crate::draw::core::geometry::rotate_geometry::RotateGeometry;
use crate::draw::types::draw_point::DrawPoint;

/// Returns the raw angle from `center` to `current_position` in radians.
#[inline]
pub fn raw_angle(current_position: DrawPoint, center: DrawPoint) -> f64 {
    RotateGeometry::angle_from_center(current_position, center)
}

/// Normalizes an angle delta to the principal range `[-pi, pi]`.
#[inline]
pub fn normalize_delta(delta: f64) -> f64 {
    RotateGeometry::normalize_delta(delta)
}

/// Snaps a total angle (`base_angle + delta`) to the nearest interval and
/// returns the snapped delta.
#[inline]
pub fn apply_discrete_snap(delta: f64, base_angle: f64, snap_interval: f64) -> f64 {
    RotateGeometry::apply_discrete_snap(delta, base_angle, snap_interval)
}
